[Tooling] Migrate Prototype Builds from App Center to Firebase App Distribution

AliSoftware opened the pull request:
## Description

This migrates the Prototype Builds to be uploaded to Firebase App Distribution instead of App Center.

This is done in the context of App Center shutting down at the end of the month.

### Internal references:

 - pdnsEh-1zD-p2 
 - paaHJt-7S2-p2
 - 186-gh-Automattic/apps-infra-plans

### Release Timing

> [!IMPORTANT]
> We might want to wait before merging this until our internal documentation—about how to use our internal tools to enroll to Firebase App Distribution—has been updated.

## Testing

 - Check that the information shown in the PR comment looks accurate, including the build number (expected to be set to the Buildkite build number) and version (expected to use the PR number)
 - Add yourself to the appropriate Testing Group via our MC tool (`/mobile/firebase-app-distribution/testers/`), then accept the invitation email you received
 - Scan the QR code in the PR comment with your iPhone and confirm you can install the prototype build on your device

AliSoftware committed:
Update gems to use FAD

AliSoftware committed:
Pipeline & Fastfile: Prototype Builds to FAD

AliSoftware committed:
Fix early return logic

AliSoftware committed:
Provide ipa_path explicitly

Since it was build in a separate job so `firebase_app_distribution` can't infer it from `gym`'s `lane_context`

dangermattic commented on fastlane/Fastfile:
```diff
@@ -97,15 +97,37 @@ platform :ios do
-    slug = 'Gravatar-SDK-Demo-UIKit'
-    upload_to_appcenter(appcenter_slug: slug, artifact_basename: 'Gravatar Demo')
+  desc 'Upload the demo app to Firebase App Distribution'
+  lane :upload_demo_to_firebase do |build_number:|
```
<table data-meta="generated_by_pr-check">
  <tbody>
    <tr>
      <td>:no_entry_sign:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/Automattic/Gravatar-SDK-iOS/blob/c50e9752fd0af31e8da0dbe688cae832e574d467/fastlane/Fastfile#L101"></span>Lint/UnusedBlockArgument: Unused block argument - <code>build_number</code>. You can omit the argument if you don't care about it.</td>
    </tr>
  </tbody>
</table>

AliSoftware committed:
Fix versioning of Prototype Builds

AliSoftware committed:
Generate new Prototype Build

Dummy empty CI commit to trigger a new Prototype Build with a new version and build number, so I can screenshot the install page from my phone for documentation purposes while having that page show more than one build (latest + previous ones)

twstokes reviewed (approved):
LGTM 🚀

AliSoftware committed:
Fix BUILDKITE_PULL_REQUEST being false not nil

AliSoftware commented:
Migration has been announced officially internally, and all the internal docs have been updated, so we're finally merging this! 🚀
